src/dcrypto/v1/v1.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package v1 implements the first version of encryption for drive
+// It uses AES-256 in CTR mode for encryption and uses authenticates
+// with an HMAC using SHA-512.
+//
+// This package should always be able to decrypt files that were encrypted
+// using this package. If there is a change that needs to be made that would
+// prevent decryption of old files, it should be done in a new version.
+
+package v1
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha512"
+	"errors"
+	"hash"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/scrypt"
+)
+
+const (
+	// The size of the HMAC sum.
+	hmacSize = sha512.Size
+
+	// The size of the HMAC key.
+	hmacKeySize = 32 // 256 bits
+
+	// The size of the random salt.
+	saltSize = 32 // 256 bits
+
+	// The size of the AES key.
+	aesKeySize = 32 // 256 bits
+
+	// The size of the AES block.
+	blockSize = aes.BlockSize
+
+	// The number of iterations to use in scrypt.Key
+	// Must be a power of 2.
+	scryptIter = 262144 // 2^18
```

Comment:
Could you please add a reference to https://godoc.org/golang.org/x/crypto/scrypt#Key as a comment before scryptIter?
This way we can refer to it. You can also disregard my comment `1<<18` since `262144` presents a better readability and comparison against the documented recommended iteration count of `interactive logins as of 2009 are N=16384` from https://godoc.org/golang.org/x/crypto/scrypt#Key